chaosgcp/apphub/actions.py

```diff
@@ -0,0 +1,295 @@
+
+
+def url_map_exists(url_map_value, project_id, region, application_name):
+    """Checks if a specific URL map value exists within AppHub-related URL maps.
+
+    This function first retrieves the selfLink URLs of all URL maps associated with services
+    in the specified AppHub application. It then searches these URL maps to determine if the
+    given `url_map_value` is present after the "urlMaps/" segment.
+
+    Args:
+        url_map_value (str): The specific value (e.g., "my-url-map") to search for after "urlMaps/" in the URL.
+        project_id (str): The ID of the Google Cloud project containing the AppHub application.
+        region (str): The region where the AppHub application is deployed.
+        application_name (str): The name of the AppHub application.
+
+    Returns:
+        bool: True if the `url_map_value` is found within the URL maps associated with
+              the specified AppHub application, False otherwise.
+    """
+    try:
+        availableUrlMaps = get_urlMap_self_links_for_matching_services(
+            apphub_app_list_services(project_id, region, application_name),
+            get_url_map_essentials(project_id),
+        )
+        for umap in availableUrlMaps:
+            index = umap.find("/urlMaps/")
+            if (
+                index != -1
+                and umap[index + len("/urlMaps/") :] == url_map_value
+            ):
+                logger.info(
+                    "given url_map is found within the URL maps associated with the specified AppHub application"
+                )
+                return True  # Found, so return True immediately
+
+    except Exception as e:
+        logger.error(f"Error retrieving URL map information: {e}")
+        return False
+
+
+def inject_fault_if_url_map_exists_app_hub(
+    application_name,
+    url_map: str,
+    project_id: str,
+    region: str,
+    target_name: str,
+    target_path: str = "/*",
+    impacted_percentage: float = 50.0,
+    http_status: int = 400,
+    regional: bool = True,
+    configuration: Configuration = None,
+    secrets: Secrets = None,
+) -> Dict[str, Any]:
+    """Processes a URL map if it exists within the specified AppHub application.
+
+    Args:
+        application_name (str): The name of the AppHub application.
+        url_map (str): The URL map value to check for.
+        project_id (str): The Google Cloud project ID.
+        region (str): The region of the AppHub application.
+        target_name (str): the the name of a path matcher in the URL map.
+        target_path (str): default "/*",
+        impacted_percentage: float default 50.0,
+        http_status (int) default 400,
+        regional: (bool) default True,
+        configuration: (Configuration) default None,
+        secrets: (Secrets) default None,
+    """
+
+    if url_map_exists(url_map, project_id, region, application_name):
+        # URL map exists, proceed with your processing logic here
+        logger.info(
```

Comment:
use debug level

Reply:
using info level , because these info are apphub  specific and required to backtrack services associated with apphub for tracing purpose in experiment .